plugins/pulp_docker/plugins/models.py

```diff
@@ -213,13 +228,63 @@ def from_json(cls, manifest_json, digest, tag, upstream_name):
                    schema_version=manifest['schemaVersion'], fs_layers=fs_layers,
                    config_layer=config_layer)
 
-    def get_symlink_name(self):
+
+class ManifestList(pulp_models.FileContentUnit, UnitMixin):
+    """
+    This model represents a Docker v2, Schema 2 Manifest list, as described here:
+
+    https://github.com/docker/distribution/blob/release/2.3/docs/spec/manifest-v2-2.md#manifest-list
+    """
+    digest = mongoengine.StringField(required=True)
+    schema_version = mongoengine.StringField(required=True)
+    manifests = mongoengine.ListField(mongoengine.StringField(), required=True)
+    amd64 = mongoengine.DictField()
+
+    # For backward compatibility
+    _ns = mongoengine.StringField(
+        default='units_{type_id}'.format(type_id=constants.MANIFEST_LIST_TYPE_ID))
+    _content_type_id = mongoengine.StringField(required=True, default=constants.MANIFEST_LIST_TYPE_ID)
+
+    unit_key_fields = ('digest',)
+
+    meta = {'collection': 'units_{type_id}'.format(type_id=constants.MANIFEST_LIST_TYPE_ID),
+            'indexes': [],
+            'allow_inheritance': False}
+
+    @classmethod
+    def from_json(cls, manifest_json, digest):
         """
-        Provides the name that should be used when creating a symlink.
-        :return: file name as it appears in a published repository
-        :rtype: str
+        Construct and return a DockerManifestList from the given JSON document.
+
+        :param manifest_json: A JSON document describing a DockerManifest object as defined by the
+                              Docker v2, Schema 2 Manifest List documentation.
+        :type  manifest_json: basestring
+        :param digest:        The content digest of the manifest, as described at
+                              https://docs.docker.com/registry/spec/api/#content-digests
+        :type  digest:        basestring
+
+        :return:              An initialized DockerManifestList object
+        :rtype:               pulp_docker.common.models.DockerManifest
         """
-        return '/'.join(('manifests', str(self.schema_version), self.digest))
+
+        mediatype = 'application/vnd.docker.distribution.manifest.v2+json'
+        manifest = json.loads(manifest_json)
+        # we will store here the digests of image manifests that manifest list contains
+        manifests = []
+        amd64 = {}
```

Comment:
Does this only ever have two keys, ``digest`` and ``schema_version`` ? If so, it's tempting to flatten this and just have two simple fields: ``amd64_digest`` and ``amd64_schema_version`` or something like that. Complex fields can cost us later in unexpected ways, as we found out with mongoengine. It's also very easy to document and understand the flattened fields.